pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTNumericLiteral.java

```diff
@@ -132,10 +132,19 @@ static int getBase(Chars image) {
             case 'b':
             case 'B':
                 return 2;
-            case '.':
-                return 10;
             default:
-                return 8;
+                if (image.indexOf('.') != -1 || image.indexOf('e') != -1 || image.indexOf('E') != -1) {
```

Comment:
I think instead of all this logic we could add a parameter `boolean isIntegral` to `getBase(Chars)`. Then in the default case we just have `return isIntegral? 8 : 10;`. In `parseIntegralValue` this value is true, we know it's integral. In `getBase()` we can just use the field `isIntegral`. This would simplify this function a lot.